be/src/exec/pipeline/scan/stream_scan_operator.cpp

```diff
@@ -0,0 +1,203 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+namespace starrocks::pipeline {
+
+
+StreamScanOperator::StreamScanOperator(OperatorFactory* factory, int32_t id, int32_t driver_sequence, int32_t dop,
+                                             ScanNode* scan_node)
+        : ConnectorScanOperator(factory, id, driver_sequence, dop, scan_node) {}
+
+void StreamScanOperator::star_epoch(const EpochInfo& epoch) {
+    is_epoch_start = true;
+    _curren_epoch = epoch;
+    // reset chunkSource status
+    // swap chunk queue
+    for (int i = 0; i < _closed_chunk_sources.size(); i++) {
+        ChunkSourcePtr chunk_source_ptr = _closed_chunk_sources.front();
+        _closed_chunk_sources.pop();
+        _closed_chunk_sources->reset_status();
+        _closed_chunk_sources->set_limit(epoch.max_rows);
+        _closed_chunk_sources->set_time_limit(epoch.max_time);
+        _old_chunk_sources.push(chunk_source_ptr)
+    }
+}
+
+bool StreamScanOperator::is_epoch_finished() {
+    if (_is_epoch_finished) {
+        return true;
+    }
+    // if storage layer returns an error, we should make sure `pull_chunk` has a chance to get it
+    if (!_get_scan_status().ok()) {
+        return false;
+    }
+
+    // Any io task is running or needs to run.
+    if (_num_running_io_tasks > 0 || !_morsel_queue->empty()) {
+        return false;
+    }
+
+    // Any shared chunk source from other ScanOperator
+    if (has_shared_chunk_source()) {
+        return false;
+    }
+
+    // Remain some data in the buffer
+    if (num_buffered_chunks() > 0) {
+        return false;
+    }
+
+    // This scan operator is finished, if no more io tasks are running
+    // or need to run, and all the read chunks are consumed.
+    return true;
+}
+
+Status StreamScanOperator::set_epoch_finishing(RuntimeState* state) {
+    std::lock_guard guard(_task_mutex);
+    _is_finished = true;
+    return Status::OK();
+}
+
+CommitOffset StreamScanOperator::get_latest_offset() {
+    // todo traverse all _old_chunks to get_latest_offset through binlog reader
+    CommitOffset offset;
+    return offset;
+}
+
+bool StreamScanOperator::is_finished() {
+    return false;
+}
+
+bool StreamScanOperatior::has_output() {
+    if (_is_finished || _is_epoch_finished) {
+        return false;
+    }
+
+    if (_is_epoch_start) {
+        _is_epoch_start = false;
+        return true;
+    }
+
+    // if storage layer returns an error, we should make sure `pull_chunk` has a chance to get it
+    if (!_get_scan_status().ok()) {
+        return true;
+    }
+
+    // Try to buffer enough chunks for exec thread, to reduce scheduling overhead.
```

Comment:
Well, since most of code is same as `ConnectorScanOperator`, please consider extending the base class instead of copy it